src/core/api/repository.go

```diff
@@ -659,6 +660,7 @@ func assembleTagsInParallel(client *registry.Repository, projectID int64, reposi
 			tag,
 			config.WithNotary(),
 			signatures,
+			limit,
```

Comment:
Just a minor suggestion, is it better to get the `true` from `limit` before starting a goroutine? In this way only 100 goroutines will be created